src/Internal/Wallet/Cip30.js

```diff
@@ -2,27 +2,32 @@
 
 exports._getNetworkId = conn => () => conn.getNetworkId();
 
-exports._getUtxos = maybe => conn => () =>
-  conn.getUtxos().then(res => (res === null ? maybe.nothing : maybe.just(res)));
+exports._getUtxos = maybe => conn => amount => paginate => () =>
+  conn
+    .getUtxos(amount, paginate)
+    .then(res => (res === null ? maybe.nothing : maybe.just(res)))
+    .catch(_ => maybe.nothing);
 
-exports._getCollateral = maybe => conn => () =>
+exports._getCollateral = maybe => conn => amount => () =>
   conn.experimental
-    .getCollateral()
+    .getCollateral({ amount: amount })
     .then(utxos =>
       utxos !== null && utxos.length ? maybe.just(utxos) : maybe.nothing
-    );
+    )
+    .catch(_ => maybe.nothing);
 
 exports._getBalance = conn => () => conn.getBalance();
 
-exports._getAddresses = conn => conn.getUsedAddresses;
+exports._getAddresses = conn => paginate => () =>
+  conn.getUsedAddresses(paginate).catch(_ => []);
 
 exports._getUnusedAddresses = conn => () => conn.getUnusedAddresses();
 
 exports._getChangeAddress = conn => () => conn.getChangeAddress();
 
 exports._getRewardAddresses = conn => () => conn.getRewardAddresses();
 
-exports._signTx = txHex => conn => () =>
+exports._signTx = txHex => _partialSign => conn => () =>
   conn.signTx(txHex, true).catch(e => {
     throw JSON.stringify(e);
   });
```

Comment:
You asked about partialSign, and how the mock can implement it. From what I understand, `partialSign` dictates (when set to true) that it's ok for there to be [required signers](https://input-output-hk.github.io/cardano-js-sdk/classes/_cardano_sdk_core.CSL.TransactionBody.html#required_signers) missing from the returned vkeys in the [witness set](https://input-output-hk.github.io/cardano-js-sdk/classes/_cardano_sdk_core.CSL.TransactionWitnessSet.html). When false, there must be no missing signers. The behaviour we already implement is if `partialSign = true` for the key wallet.